src/encoded/commands/load_users_from_oc.py

```diff
@@ -0,0 +1,152 @@
+#   SMaHT Listed First Name,
+#   Email,
+#   SMaHT Contact PI,
+#   Association,
+#   Grant Component,
+#   DAC code in the portal
+
+# Define the named tuple
+User = namedtuple('User', ['first_name', 'last_name', 'email', 'submission_center'])
+
+
+class UserCSVProcessorException(Exception):
+    pass
+
+
+class UserCSVProcessor:
+
+    def __init__(self, env='data'):
+        self.key = SMaHTKeyManager().get_keydict_for_env(env)
+        self.submission_centers = []
+        self.user_dict = {}
+        self.validate_only = False
+
+    def read_csv(self, file_path: str) -> list:
+        """ Pulls the whole CSV into memory and returns a list of rows """
+        with open(file_path, 'r') as csv_file:
+            csv_reader = csv.reader(csv_file)
+            user_list = [row for row in csv_reader if any(row)]  # strip empties
+        return user_list
+
+    @staticmethod
+    def build_user_from_row(row: list) -> User:
+        """ Builds a 'User' namedtuple extracting from the format above """
+        first_name, last_name, email, submission_center = row[2], row[1], row[3], row[6]
+        return User(first_name, last_name, email, submission_center)
+
+    def generate_submission_center_list(self, user_csv_list: list[list]):
+        """ Goes through the CSV and populates the submission center list """
+        for row in user_csv_list:
+            sc = row[6]
+            if sc not in self.submission_centers:
+                self.submission_centers.append(sc)
+
+    def validate_submission_center_list(self):
+        """ Validates all submission centers exist on the portal """
+        if not self.submission_centers:
+            raise UserCSVProcessorException(f'Attempted to validate submission centers prior to loading them')
+        for sc in self.submission_centers:
+            if sc == 'dac':  # XXX: Hardcode as this is not named correctly
+                sc = 'smaht_dac'
+            elif sc == 'nih':  # XXX: Hardcode as NIH has no submission center
+                continue
+            ff_utils.get_metadata(f'/submission-centers/{sc.lower()}', key=self.key)  # this will throw exception if not found
+
+    def check_for_existing_user(self, user: User) -> bool:
+        """ Checks if the current user already exists """
+        email = user.email
+        user = ff_utils.get_metadata(f'/{email}', key=self.key)
+        if not user:
+            return False
+        return True
+
+    def generate_users(self, user_csv_list: list[list]) -> dict:
+        """ Generates an email --> props mapping of users to post """
+        for u in user_csv_list:
+            user = self.build_user_from_row(u)
+            if user.email in self.user_dict:
+                raise UserCSVProcessorException(f'Found duplicate user in spreadsheet: {user.email}')
+            self.user_dict[user.email] = user
+        return self.user_dict
+
+    def remove_existing_users(self):
+        """ Strips out users who already have a user record """
```

Comment:
I'd've just named this something like ignore_existing_users just because at first glance I'd thought this was going to remove the existing users from the database, rather than marking for ignore, the users in the list which are already in the database - though the comment obviously does say this.